ingestion_server/ingestion_server/ingest.py

```diff
@@ -0,0 +1,266 @@
+        password=UPSTREAM_DB_PASSWORD,
+        host=UPSTREAM_DB_HOST,
+        connect_timeout=5
+    )
+    query_cols = ','.join(_get_shared_cols(downstream_db, upstream_db, table))
+    upstream_db.close()
+    # Connect to upstream database and create references to foreign tables.
+    log.info('(Re)initializing foreign data wrapper')
+    init_fdw = '''
+        CREATE EXTENSION IF NOT EXISTS postgres_fdw;
+        DROP SERVER IF EXISTS upstream CASCADE;
+        CREATE SERVER upstream FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host '{host}', dbname 'openledger', port '{port}');
+
+        CREATE USER MAPPING IF NOT EXISTS FOR deploy SERVER upstream
+        OPTIONS (user 'deploy', password '{passwd}');
+        DROP SCHEMA IF EXISTS upstream_schema CASCADE;
+        CREATE SCHEMA upstream_schema AUTHORIZATION deploy;
+
+        IMPORT FOREIGN SCHEMA public
+        LIMIT TO ({table}) FROM SERVER upstream INTO upstream_schema;
+    '''.format(host=UPSTREAM_DB_HOST, passwd=UPSTREAM_DB_PASSWORD, table=table,
+               port=UPSTREAM_DB_PORT)
+    # 1. Import data into a temporary table
+    # 2. Recreate indices from the original table
+    # 3. Recreate constraints from the original table.
+    # 4. Delete orphaned foreign key references.
+    # 5. Promote the temporary table and delete the original.
+    copy_data = '''
+        DROP TABLE IF EXISTS temp_import_{table};
+        CREATE TABLE temp_import_{table} (LIKE {table} INCLUDING CONSTRAINTS);
+        INSERT INTO temp_import_{table} ({cols})
+        SELECT {cols} from upstream_schema.{table};
+        DROP SERVER upstream CASCADE;
+    '''.format(table=table, cols=query_cols)
+    create_indices = ';\n'.join(_generate_indices(downstream_db, table))
+    remap_constraints = ';\n'.join(_generate_constraints(downstream_db, table))
+    go_live = '''
+        DROP TABLE {table};
+        ALTER TABLE temp_import_{table} RENAME TO {table};
+    '''.format(table=table)
+
+    with downstream_db.cursor() as downstream_cur:
+        log.info('Copying upstream data...')
+        downstream_cur.execute(init_fdw)
+        downstream_cur.execute(copy_data)
+        log.info('Copying finished! Recreating database indices...')
+        _update_progress(progress, 50.0)
+        if create_indices != '':
+            downstream_cur.execute(create_indices)
+        _update_progress(progress, 70.0)
+        log.info('Done creating indices! Remapping constraints...')
+        if remap_constraints != '':
+            downstream_cur.execute(remap_constraints)
+        _update_progress(progress, 99.0)
+        log.info('Done remapping constraints! Going live with new table...')
+        downstream_cur.execute(go_live)
+    downstream_db.commit()
+    downstream_db.close()
+    log.info('Finished refreshing table \'{}\'.'.format(table))
+    _update_progress(progress, 100.0)
+    if finish_time:
+        finish_time.value = datetime.datetime.utcnow().timestamp()
+    return
```

Comment:
My thinking at the time was probably "explicit is better than implicit", but I can understand how this could be seen as needless clutter. I've removed it.